### ISSUE TITLE
[Docker] Execute web application fail while upgrade from v2.4.0 to 2.8.1

### ISSUE THREAD

**Issue by kobayashilin** [SourceLabOrg/kafka-webview#254]
Hi,

Thanks for quickly upgrade the security patch #244 ~!! It's awesome!!!

However, we encounter some problem

```
2021-12-17 06:58:51.370  INFO 1 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 5369 ms
2021-12-17 06:58:52.061  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2021-12-17 06:58:52.607  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2021-12-17 06:58:52.677  INFO 1 --- [           main] o.f.c.internal.license.VersionPrinter    : Flyway Community Edition 5.2.4 by Boxfuse
2021-12-17 06:58:52.701  INFO 1 --- [           main] o.f.c.internal.database.DatabaseFactory  : Database: jdbc:h2:file:./data/db (H2 1.4)
2021-12-17 06:58:52.771  WARN 1 --- [           main] o.f.c.internal.database.base.Database    : Flyway upgrade recommended: H2 1.4.200 is newer than this version of Flyway and support has not been tested.
2021-12-17 06:58:52.954 ERROR 1 --- [           main] o.s.b.web.embedded.tomcat.TomcatStarter  : Error starting Tomcat context. Exception: org.springframework.beans.factory.UnsatisfiedDependencyException. Message: Error creating bean with name 'securityConfig': Unsatisfied dependency expressed through field 'userRepository'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'userRepository': Cannot create inner bean '(inner bean)#4cf8b2dc' of type [org.springframework.orm.jpa.SharedEntityManagerCreator] while setting bean property 'entityManager'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name '(inner bean)#4cf8b2dc': Cannot resolve reference to bean 'entityManagerFactory' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Invocation of init method failed; nested exception is org.flywaydb.core.api.FlywayException: Validate failed: Migration checksum mismatch for migration version 1
-> Applied to database : -547719129
-> Resolved locally    : 103390073
```

docker-compose file
```
version: "3"
services:
  kafka-webviewer-dev:
    image: sourcelaborg/kafka-webview:2.8.1
    ports:
      - 18080:8080
    volumes:
      - ./data:/app/data
    restart: always
```

OS: Ubuntu
ls -al volumes
```
drwxrwxr-x 2 ubuntu ubuntu  4096 Jul 24  2019 .
drwxrwxr-x 3 ubuntu ubuntu  4096 Dec 17 07:07 ..
-rw-r--r-- 1 root   root   65536 Dec 17 07:00 db.mv.db
```

If there is any problem or miss configuration, please let me know. 

Thanks!

Yan-Ting

**Comment by Crim** [comment]
Hmm haven't seen that before.  Which version of Kafka-webview were you running prebviously?  I'm wondering if it's an update to the internal H2 database library that perhaps changed between versions.

**Comment by kobayashilin** [comment]
> Hmm haven't seen that before. Which version of Kafka-webview were you running prebviously? I'm wondering if it's an update to the internal H2 database library that perhaps changed between versions.

Thanks for the reply.
I change the version from v2.4.0 to 2.8.1 on the docker-compose.yml file

**Comment by Crim** [comment]
Spent some time digging into this and it appears that at some point I changed the whitespacing on one of the database migration files, which results in this unfortunate error.

The "quick" but likely not acceptable fix is to delete your data volume and start with a fresh database.  This unfortunately will mean you'll need to rebuild any saved configuration, and likely not a good solution.

The other fix is to connect to the H2 database and issue the following query:

```sql
update  "flyway_schema_history" set "checksum"=103390073 where "version"=1;
```

I was able to do this by running the following docker image:
```
run -d -p 1521:1521 -p 81:81 -v kafkawebview_data:/opt/h2-data --name=MyH2Instance oscarfonts/h2
```

Then connecting to `http://localhost:81` and changing the database name to `db`
<img width="469" alt="image" src="https://user-images.githubusercontent.com/571653/146661471-f38cbffa-336a-4359-9641-b3a36d747514.png">

And executing the query within that console.  There is likely otherways to connect to the H2 database....

**Comment by Crim** [comment]
Ok, I've released 2.8.2 which disables the checksum validations, which should resolve your problem w/o any manual steps on your part.

Let me know if things start up w/o any errors after trying that.

Thanks
Stephen

**Comment by kobayashilin** [comment]
> Ok, I've released `v2.8.2` which disables the checksum validations, which should resolve your problem w/o any manual steps on your part.
> 
> Let me know if things start up w/o any errors after trying this release version.
> 
> Thanks Stephen

It works! Thanks for the help. 

![image](https://user-images.githubusercontent.com/4756036/146699745-d34bfd03-7f3b-4bba-8751-13cb834aab29.png)

Wish you have a happy Christmas week :)

Best regards,
Yan-Ting
